### ISSUE TITLE
ScaledObject.keda.k8s.io is invalid: spec.maxReplicaCount in body must be of type integer

### ISSUE THREAD

**Issue by jldec** [projectriff/system#134]
I'm seeing this recurring error in the log from the streaming-controller-manager when i wire a simple echo function as a stream processor from an input to an output stream.

The function appears to be operating properly on the stream despite the error which repeats in the log every few minutes.

~/riff/echo/echo.js
```js
module.exports = x => {
  console.log('echo', x);
  return x;
}
```

create `echo` function
```sh
riff function create echo \
  --local-path ~/riff/echo \
  --artifact echo.js \
  --tail
```

wire `in` and `out` streams via a `test` processor
```sh
riff streaming stream create in  --provider franz-kafka-provisioner --content-type application/json 
riff streaming stream create out --provider franz-kafka-provisioner --content-type application/json 

riff streaming processor create test \
  --function-ref echo \
  --input in \
  --output out
```

error from `kubectl logs -f riff-streaming-controller-manager-... -c manager -n riff-system`
```sh
ERROR	controller-runtime.controller	Reconciler error	{"controller": "processor", "request": "default/test", "error": "ScaledObject.keda.k8s.io \"test-processor\" is invalid: []: Invalid value: map[string]interface {}{\"apiVersion\":\"keda.k8s.io/v1alpha1\", \"kind\":\"ScaledObject\", \"metadata\":map[string]interface {}{\"creationTimestamp\":\"2019-10-17T11:54:01Z\", \"generation\":1, \"labels\":map[string]interface {}{\"deploymentName\":\"test-processor\", \"streaming.projectriff.io/processor\":\"test\"}, \"name\":\"test-processor\", \"namespace\":\"default\", \"ownerReferences\":[]interface {}{map[string]interface {}{\"apiVersion\":\"streaming.projectriff.io/v1alpha1\", \"blockOwnerDeletion\":true, \"controller\":true, \"kind\":\"Processor\", \"name\":\"test\", \"uid\":\"69081ea3-f0d4-11e9-a9b7-025000000001\"}}, \"uid\":\"cf393560-f0d4-11e9-a9b7-025000000001\"}, \"spec\":map[string]interface {}{\"cooldownPeriod\":30, \"maxReplicaCount\":interface {}(nil), \"minReplicaCount\":interface {}(nil), \"pollingInterval\":1, \"scaleTargetRef\":map[string]interface {}{\"containerName\":\"\", \"deploymentName\":\"test-processor\"}, \"triggers\":[]interface {}{map[string]interface {}{\"metadata\":map[string]interface {}{\"address\":\"franz-kafka-liiklus.default:6565\", \"group\":\"test\", \"topic\":\"default_in\"}, \"name\":\"\", \"type\":\"liiklus\"}}}, \"status\":map[string]interface {}{\"currentReplicas\":0, \"desiredReplicas\":0}}: validation failure list:\nspec.maxReplicaCount in body must be of type integer: \"null\"\nspec.minReplicaCount in body must be of type integer: \"null\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/home/runner/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:218
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:192
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.2/pkg/internal/controller/controller.go:171
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/home/runner/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88
```

**Comment by ericbottard** [comment]
Can you confirm you're pre- https://github.com/projectriff/system/commit/52022e024a4d95a944b8f2261148b5dc9002cd2c ?
This should have been fixed by now

**Comment by scothis** [comment]
@jldec FYI, you can add an issue to the streaming project by adding the `streaming-runtime` label on this issue.

**Comment by ericbottard** [comment]
Fixed by 52022e0
